[FLINK-26295][table] Operation strategies should allow having spaces …

snuyanzin opened the pull request:
## What is the purpose of the change

The PR allows to have spaces between sql client command and semicolon like it was before https://github.com/apache/flink/pull/18363

e.g. 
```sql
HELP ;
QUIT ;
```
Now it fails as described in FLINK-26295
## Brief change log

Added spaces support in Operations

## Verifying this change
There is a new test _org.apache.flink.table.planner.operations.SqlToOperationConverterTest#testSqlClientCommands_

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: ( no )
  - The runtime per-record code paths (performance sensitive): ( no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no )
  - The S3 file system connector: ( no )

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable)

slinkydeveloper commented on flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/parse/ClearOperationParseStrategy.java:
```diff
@@ -29,7 +29,7 @@
 
     private ClearOperationParseStrategy() {
-        super(Pattern.compile("CLEAR;?", DEFAULT_PATTERN_FLAGS));
+        super(Pattern.compile("CLEAR(\\s*;)?", DEFAULT_PATTERN_FLAGS));
```
What about just using `CLEAR\s*;?`. This way you check any eventual spaces, but not a space after the `;` (which should be the beginning of another command)

slinkydeveloper replied:
Same comments for all the ones below

snuyanzin replied:
done, thanks

snuyanzin commented:
@flinkbot run azure

slinkydeveloper commented on flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java:
```diff
@@ -1660,6 +1665,65 @@ public void testReset() {
+    @Test
+    public void testQuitCommands() {
+        ExtendedParser extendedParser = new ExtendedParser();
+        Stream.of(
```
Please create a parametrized test for this. It's quite easy in JUnit 5, just use `@ValueSource`: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-ValueSource

slinkydeveloper replied:
I gave a quick look to this class, porting to junit5 should be as easy as just replacing the imports. Also, I think it's possible for JUnit 4 and 5 to coexist in the same class (but i'm not 100% sure)

snuyanzin replied:
done

slinkydeveloper commented on flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/operations/SqlToOperationConverterTest.java:
```diff
@@ -1660,6 +1665,65 @@ public void testReset() {
+                            Operation operation1 =
+                                    extendedParser
+                                            .parse(command)
+                                            .orElseThrow(
```
You can remove the `orElseThrow` with a proper assertj assertion:

```
assertThat(extendedParser.parse(command))
  .get() // Returns an assertion on the value, and throws proper AssertionException if no value is present
  .isInstanceOf(QuitOperation.class);
```

snuyanzin replied:
thanks, done

snuyanzin commented:
@flinkbot run azure

snuyanzin commented:
@slinkydeveloper thank you for your feedback, I addressed all the comments.
Please have a look once you have time.
Please let me know if I need to squash commits
